### ISSUE TITLE
(Epic) All current components comply with accessibility standards (WCAG 2.0 and WCAG 2.1)

### ISSUE THREAD

**Issue by el7h1rw2** [Royal-Navy/design-system#809]
"The deadline for public sector organisations to make all existing websites accessible is 23 September 2020, and the compliance date for mobile applications is 23 June 2021."

**Resources:**

- https://www.gov.uk/government/news/regulations-on-the-accessibility-of-new-public-sector-websites-come-into-force
- https://webaim.org/standards/wcag/WCAG2Checklist.pdf

**Work Items:**
- [x] #828 Add Storybook Accessibility add-on
- [x] #903 (Spike) Investigate A11Y requirements for Design System
- [x] #952 Run aXe against docs site
- [x] #953 Fix violations flagged by a11y add on
- [x] #954 Spike how aria attributes can be utilised across the design system
- [x] #955 Create accessibility statement
- [x] #956 Integrate a11y checks with CI
- [x] #1059 Add aria attributes to Alert component
- [x] #1061 Add aria attributes to Button component
- [x] #1060 Add aria attributes to Breadcrumbs component
- [x] #1062 Add aria attributes to ButtonGroup component
- [x] #1063 Add aria attributes to DataList component 
- [x] #1064 Add aria attributes to DatePicker component
- [x] #1065 Add aria attributes to Masthead component
- [x] #1066 Add aria attributes to Dialog/Modal components
- [x] #1067 Add aria attributes to NumberInput component
- [x] #1068 Add aria attributes to Pagination component
- [x] #1069 Add aria attributes to Popover component 
- [x] #1070 Add documentation for ProgressIndicator accessibility
- [x] #1071 Add aria attributes to RangeSlider 
- [x] #1074 Add aria attributes to Sidebar
- [x] #1075 Add aria attributes to Table
- [x] #1076 Add aria attributes to TabSet
- [x] #1077 Add aria attributes to Timeline
- [x] #1078 Add aria attributes to Toast
- [x] #1079 Add aria attributes to Tooltip
- [x] #1097 Data List a11y violations
- [x] #1098 Dropdown a11y violations
- [x] #1101 Number Input a11y violations
- [x] #1100 Nav a11y violations
- [x] #1103 TabSet a11y violations
- [x] #1104 Text Input a11y violations 
- [x] #1261 Sort icons in `Table` need `aria-hidden` 
- [x] #1262 Add aria attributes to `Masthead` avatar sheet
- [x] #1263 Add aria attributes to `List` component
- [x] #1264 Add `caption` to `Table` 
- [x] #1268 Add aria attributes to TimelineSide

**Comment by el7h1rw2** [comment]
We need to conduct accessibility testing on the components and documentation site. I've also been provided with these attachments:

JSP912_Part2.pdf
Defstan 00-251 Part 3.pdf
and the following link: http://jsp604.r.mil.uk/index.php/Accessibility_in_ICT_Projects

This mandates WCAG 2.0 ICT AA and WCAG 2.1 AA as a minimum, but there are lots of other useful links.

I’ve downloaded several of the Defence Standards, but they all seem very high level and more about the process of defining and running a contract.

As far as the other ISO 25010 NFRs goes, I think we need to just agree what is sensible from a Nelson perspective. Could I get some support from the team to flesh these out next week?

**Comment by m7kvqbe1** [comment]
https://accessibility.campaign.gov.uk/
